manifests/user.pp

```diff
@@ -13,63 +13,94 @@
+      default: { fail('invalid privilege level specified') }
+    }
 
-  exec { "ipmi_user_enable_${title}":
-    command     => "/usr/bin/ipmitool user enable ${user_id}",
-    refreshonly => true,
-  }
+    if $password =~ Sensitive {
+      # unwrap before Puppet 6.24 can only be called on Sensitive values
+      $real_password = $password.unwrap
+    } else {
+      $real_password = $password
+    }
 
-  exec { "ipmi_user_add_${title}":
-    command => "/usr/bin/ipmitool user set name ${user_id} ${user}",
-    unless  => "/usr/bin/test \"$(ipmitool user list 1 | grep '^${user_id}' | awk '{print \$2}' | head -n1)\" = \"${user}\"",
-    notify  => [Exec["ipmi_user_priv_${title}"], Exec["ipmi_user_setpw_${title}"]],
-  }
+    exec { "ipmi_user_enable_${title}":
+      command     => "/usr/bin/ipmitool user enable ${user_id}",
+      refreshonly => true,
+    }
 
-  exec { "ipmi_user_priv_${title}":
-    command => "/usr/bin/ipmitool user priv ${user_id} ${priv} 1",
-    unless  => "/usr/bin/test \"$(ipmitool user list 1 | grep '^${user_id}' | awk '{print \$6}' | head -n1)\" = ${privilege}",
-    notify  => [Exec["ipmi_user_enable_${title}"], Exec["ipmi_user_enable_sol_${title}"], Exec["ipmi_user_channel_setaccess_${title}"]],
-  }
+    exec { "ipmi_user_add_${title}":
+      command => "/usr/bin/ipmitool user set name ${user_id} ${user}",
+      unless  => "/usr/bin/test \"$(ipmitool user list 1 | grep '^${user_id}' | awk '{print \$2}' | head -n1)\" = \"${user}\"",
+      notify  => [Exec["ipmi_user_priv_${title}"], Exec["ipmi_user_setpw_${title}"]],
+    }
 
-  exec { "ipmi_user_setpw_${title}":
-    command => "/usr/bin/ipmitool user set password ${user_id} \'${real_password}\'",
-    unless  => "/usr/bin/ipmitool user test ${user_id} 16 \'${real_password}\'",
-    notify  => [Exec["ipmi_user_enable_${title}"], Exec["ipmi_user_enable_sol_${title}"], Exec["ipmi_user_channel_setaccess_${title}"]],
-  }
+    exec { "ipmi_user_priv_${title}":
+      command => "/usr/bin/ipmitool user priv ${user_id} ${priv} 1",
+      unless  => "/usr/bin/test \"$(ipmitool user list 1 | grep '^${user_id}' | awk '{print \$6}' | head -n1)\" = ${privilege}",
+      notify  => [Exec["ipmi_user_enable_${title}"], Exec["ipmi_user_enable_sol_${title}"], Exec["ipmi_user_channel_setaccess_${title}"]],
+    }
 
-  exec { "ipmi_user_enable_sol_${title}":
-    command     => "/usr/bin/ipmitool sol payload enable 1 ${user_id}",
-    refreshonly => true,
-  }
+    exec { "ipmi_user_setpw_${title}":
+      command => "/usr/bin/ipmitool user set password ${user_id} \'${real_password}\'",
+      unless  => "/usr/bin/ipmitool user test ${user_id} 16 \'${real_password}\'",
+      notify  => [Exec["ipmi_user_enable_${title}"], Exec["ipmi_user_enable_sol_${title}"], Exec["ipmi_user_channel_setaccess_${title}"]],
+    }
+
+    exec { "ipmi_user_enable_sol_${title}":
+      command     => "/usr/bin/ipmitool sol payload enable 1 ${user_id}",
+      refreshonly => true,
```

Comment:
Do the refershonly exec resources need to be inside of the `$enable` if?  There is a lot going on here and I'm wondering if cutting down the size of the if blocks whould make it easier to understand the logic.

Reply:
I was thinking keeping them within the `if` scope helped make it clear these elements are part of the same logical block.  I'm open to moving them out, but having them under the same `{}` seemed to be a clean blob.